README.md

```diff
@@ -92,13 +100,77 @@ matches the user UID or name which is running inside the container.**
 Plugins
 ---------------------------------
 
+### Base set of plugins
+
+An initial set of Jenkins plugins are included in the OpenShift Jenkins images.  The general methodology 
+is that the CentOS7 image if first updated with any changes to the list of plugins.  After some level 
+of verification with that image, the RHEL7 image is updated.
+
+#### Plugin installation for CentOS7
+
+The top level list of plugins to install is located [here](blob/master/2/contrib/openshift/base-plugins.txt).  The
+format of the file is:
+
+```
+pluginId:pluginVersion
+```
+
+The file is processed by the following call in the [CentOS7 Dockerfile](blob/master/2/Dockerfile):
+
+```
+/usr/local/bin/install-plugins.sh /opt/openshift/base-plugins.txt
+```
+
+Where both [base-plugins.txt](blob/master/2/contrib/openshift/base-plugins.txt) and [install-plugins.sh](blob/master/2/contrib/jenkins/install-plugins.sh)
+are copied into the image prior to that invocation.
+
+The running of `install-plugins.sh` will download the files listed in `base-plugins.txt`, and then open each plugin's manifest
+and download any needed dependencies listed, including upgrading any previously installed dependencies as needed.
+
+To update the version of a plugin or add a new plugin, construct a PR for this repository that updates `base-plugins.txt` appropriately.
+Administrators for this repository will make sure necessary tests are run and merge the PR when things are ready.
+
+When PRs for this repository are merged, they kick off associated builds in the [`push_jenkins_images` job on OpenShift's public 
+Jenkins CI/CD server](https://ci.openshift.redhat.com/jenkins/view/All/job/push_jenkins_images/).  When those build complete,
+new versions of the CentOS7 based versions of the images produced by this repository are pushed to Docker HUB.  See the top of the README
+for the precise list.
+
+#### Plugin installation for RHEL7
+
+Only OpenShift developers working for Red Hat can update the list of plugins for the RHEL7 image.  For those developers, visit this
+[internal Jenkins server](https://buildvm.openshift.eng.bos.redhat.com:8443/job/devex/job/devex%252Fjenkins-plugins/)
+and log in (contact our CD team for permissions to this job).  Click the `Build with parameters` link, update the `PLUGIN_LIST` field, 
+and submit the build.  The format of the data for the `PLUGIN_LIST` field is the same as `base-plugins.txt`.
+
+The complete list of plugins needs to be provided though.  The most straight forward approach is to mine the output of the 
```

Comment:
"The complete list of plugins (ie including dependencies) needs to be provided."